### PR TITLE
Add and integrate Snapdragon Game Super Resolution(SGSR) into js sdk

### PR DESCRIPTION
**Pull request by adglkh**
Snapdragon™ Game Super Resolution(SGSR), which integrates upscaling
and sharpening in one single GPU shader pass. The algorithm uses a
12-tap Lanczos-like scaling filter and adaptive sharpening filter,
which presents smooth images and sharp edges. For more details:
    
     https://github.com/SnapdragonStudios/snapdragon-gsr
    
This change adds and integrates SGSR, this single GPU shader pass
into the streaming js sdk, and usd by default for the time being.
    
NOTE: for now, FSR is still kept in the source for reference.
In the end, we will choose the best upscaling algorithm from all
available options after benchmarking and comparing performance and
quality, and only ship the best one in the streaming SDK.

### CODE REVIEW

**Comment by adglkh** [comment]
Let's hold off on merging this until we've discussed how to manage multiple shaders in the main branch or decide whether to keep FSR before we reach a conclusion after benchmarking and comparing the performance and quality of all available upscaling algorithms (FSR, Anime4K, SGSR), choose the best one, and use it in our JS SDK.

**Review by morphis — CHANGES_REQUESTED** [review]
Can we split out SGSR integration into an example? That way we can merge and then defer the decision to a later point. Wdyt?

**Comment by morphis** [comment]
@adglkh we don't want to switch the actual shader implementation in the SDK yet. So what we can do instead is write an example application using SGSR and add that to https://github.com/canonical/anbox-streaming-sdk/tree/main/examples/js as part of this PR. Then we have an example demonstrating the use and have no pending PR.

**Comment by adglkh** [comment]
> @adglkh we don't want to switch the actual shader implementation in the SDK yet. So what we can do instead is write an example application using SGSR and add that to https://github.com/canonical/anbox-streaming-sdk/tree/main/examples/js as part of this PR. Then we have an example demonstrating the use and have no pending PR.

A: okay, that makes sense to me.

**Comment by adglkh** [comment]
The image is broken, which has blocked e2e testing. Let's wait for the new image to be built and cached today in the next run.

**Review by morphis — APPROVED** [review]
LGTM

File: examples/js/sgsr.html [hunk 1]
```diff
@@ -0,0 +1,218 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <meta content="text/html;charset=utf-8" http-equiv="Content-Type">
+    <meta content="utf-8" http-equiv="encoding">
+    <title>Anbox Streaming SDK Example - Enable video upscaling with Snapdragon™ Game Super Resolution(SGSR) </title>
```
Comment (ajanon) [inline examples/js/sgsr.html:6]: Random question: do we need an agreement from them to use this name?
Or put a disclaimer (e.g. 'This trademark is possessed by...')

**Review by ajanon — APPROVED** [review]
LGTM, just one question

File: examples/js/sgsr.html (hunk 1 above)
Comment (adglkh) [reply examples/js/sgsr.html:6]: Thanks @ajanon , would you mind taking another look?

**Review by ajanon — APPROVED** [review]
LGTM

**Pull request merged**